app/Models/License.php

```diff
@@ -175,12 +176,24 @@ public static function adjustSeatCount($license, $oldSeats, $newSeats)
             return true;
         }
         // Else we're adding seats.
-        DB::transaction(function () use ($license, $oldSeats, $newSeats) {
-            for ($i = $oldSeats; $i < $newSeats; $i++) {
-                $license->licenseSeatsRelation()->save(new LicenseSeat, ['user_id' => Auth::id()]);
-            }
+        //Create enough seats for the change.
+        $licenseInsert = [];
+        for ($i = $oldSeats; $i < $newSeats; $i++) {
+            $licenseInsert[] = [
+                'user_id' => Auth::id(),
+                'license_id' => $license->id,
+                'created_at' => now(),
+                'updated_at' => now()
+            ];
+        }
+        //Chunk and use DB transactions to prevent timeouts.
```

Comment:
Huh, okay, maybe this is what we're doing instead? a cleverly-collected chunked thing? I could easily get behind that.

Reply:
Yeah, basically instead of letting Eloquent manage all of those inserts with all of it's overhead. We are just creating a few big inserts using the QueryBuilder.